packages/cms-data-sync/src/interest-groups/interest-groups.data-migration.ts

```diff
@@ -0,0 +1,135 @@
+
+type InterestGroupItem = NonNullable<
+  NonNullable<FetchGroupsQuery['queryGroupsContentsWithTotal']>['items']
+>[number];
+
+export const migrateInterestGroups = async () => {
+  const { contentfulEnvironment, squidexGraphqlClient } =
+    await getSquidexAndContentfulClients();
+
+  const migrateFromSquidexToContentful = migrateFromSquidexToContentfulFactory(
+    contentfulEnvironment,
+    logger,
+  );
+
+  const fetchAllData = async (): Promise<InterestGroupItem[]> =>
+    paginatedFetch<InterestGroupItem>(async (take: number, skip: number) => {
+      const { queryGroupsContentsWithTotal } =
+        await squidexGraphqlClient.request<
+          FetchGroupsQuery,
+          FetchGroupsQueryVariables
+        >(interestGroupsQuery, { take, skip });
+      return queryGroupsContentsWithTotal;
+    });
+
+  const fetchData = async (): Promise<InterestGroupItem[]> => {
+    const entries = await fetchAllData();
+    logger(`Loaded ${entries.length} entries...`, 'INFO');
+    return entries;
+  };
+
+  const parseInterestGroupItem = async (group: InterestGroupItem) => {
+    const { id, flatData } = group;
+    const { teams, calendars, leaders, tools, thumbnail, ...props } = flatData;
+
+    const leaderLinks = await Promise.all(
+      (leaders || []).map(async (leader) => {
+        if (leader.user && leader.user[0]) {
+          const leaderEntry = await contentfulEnvironment.createEntry(
+            'interestGroupLeaders',
+            {
+              fields: addLocaleToFields({
+                role: leader.role,
+                inactiveSinceDate: leader.inactiveSinceDate,
+                user: {
+                  sys: {
+                    type: 'Link',
+                    linkType: 'Entry',
+                    id: leader.user[0].id,
+                  },
+                },
+              }),
+            },
+          );
+          await leaderEntry.publish();
+          return {
+            sys: {
+              type: 'Link',
+              linkType: 'Entry',
+              id: leaderEntry.sys.id,
+            },
+          };
+        }
+        return null;
+      }),
+    );
+
+    const teamLinks = (teams || []).map((team) => {
+      if (team.id) {
+        return {
+          sys: {
+            type: 'Link',
+            linkType: 'Entry',
+            id: team.id,
+          },
+        };
+      }
+      return null;
+    });
+
+    let thumbnailAsset;
+    if (thumbnail) {
+      const urls = createAssetUrl(thumbnail.map((asset) => asset.id));
+      if (urls.length > 0) {
+        thumbnailAsset = await createAsset(contentfulEnvironment, [
+          { ...thumbnail[0], thumbnailUrl: urls[0] },
+        ]);
+      }
+    }
+
+    let calendar;
+    if (calendars && calendars.length && calendars[0].id) {
+      calendar = {
+        sys: {
+          type: 'Link',
+          linkType: 'Entry',
+          id: calendars[0].id,
+        },
+      };
```

Comment:
It shouldn't happen if migrated in the right order, but in the events migration https://github.com/yldio/asap-hub/blob/CT-14/interest-group-migration/packages/cms-data-sync/src/events/events.data-migration.ts#L190 I've checked if the calendar exists first then I would create a link. I don't think this is super necessary, but I just wonder what would happen if we link to a calendar that does not exist.